src/client.ts

```diff
@@ -425,14 +432,49 @@ in an environmental variable named FAUNA_SECRET or pass it to the Client\
       headerObject["x-last-txn-ts"] = this.#lastTxnTs;
     }
   }
+
+  #validateConfiguration() {
+    const config = this.#clientConfiguration;
+    if (config.client_timeout_ms) {
+      if (!config.query_timeout_ms) {
+        throw new TypeError(
+          `Expected number, but received ${config.query_timeout_ms}. 'query_timeout_ms' must be configured if 'client_timeout_ms' is configured.`
+        );
+      }
```

Comment:
Default is "there is no query_timeout_ms". Our Jira says that default is to not provide a query timeout, so that core can handle the default.

This goes away though if we have change to the concept of a configured buffer, rather than a configured absolute value.